File: 187-repeated-dna-sequences/repeated-dna-sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {

        HashSet<String> set = new HashSet<>();
        HashSet<String> repeat = new HashSet<>();
        List<String> ans = new ArrayList<>();

        if(s.length()<10){
            return ans;
        }

        String str = s.substring(0, 10);
        set.add(str);

        for (int i = 10; i < s.length(); i++) {
            str += s.charAt(i);
            str = str.substring(1);

            if (set.contains(str))
                repeat.add(str);
            else
                set.add(str);
        }
        ans.addAll(repeat);
        return ans;
    }
}

File: 187-repeated-dna-sequences/repeated-dna-sequences.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):

        repeat_set = set()
        check_set = set()

        for i in range(len(s) - 9):

            temp_str = s[i:i + 10]

            if temp_str in check_set:
                repeat_set.add(temp_str)
            else:
                check_set.add(temp_str)

        return list(repeat_set)